### PR TITLE
Renew install / upgrade system

### PR DESCRIPTION
**Pull request by btry**
solve #453 

This PR assumes #479 was merged.

This PR needs update after #477 validation and merge

### CODE REVIEW

**Comment by btry** [comment]
Trying to introduce unit tests for schema checks, I found some inconsistencies. Will fix very soon.

**Comment by btry** [comment]
Unit tests now check the tables are the same as the .sql file. (view excluded from the test).

File: inc/form.class.php [hunk 1]
```diff
@@ -958,6 +958,12 @@ public function prepareInputForAdd($input)
          $input['content'] = addslashes($input['content']);
       }
 
+      if (!isset($input['requesttype'])) {
```
Comment (orthagh) [inline inc/form.class.php]: it's not a requested change, but what's the point of this change in this pr ?

File: setup.php [hunk 2]
```diff
@@ -1,7 +1,10 @@
 <?php
 global $CFG_GLPI;
 // Version of the plugin
-define('PLUGIN_FORMCREATOR_VERSION', "2.5");
+define('PLUGIN_FORMCREATOR_VERSION', "2.5.0");
+// Schema version of this version
+define('PLUGIN_FORMCREATOR_SCHEMA_VERSION', "2.5");
```
Comment (orthagh) [inline setup.php]: 2.5.0 ?

File: inc/form.class.php (hunk 1 above)
Comment (btry) [inline inc/form.class.php]: Have a look into PluginFormcreatorForm::install() in previous  versions : request type has a default value computed at install time and is not set when creating a form.

I think this is not a good design. 

```
         // Create Forms table
         $query = "CREATE TABLE IF NOT EXISTS `$table` (
                     `id` int(11) NOT NULL AUTO_INCREMENT PRIMARY KEY,
                     `entities_id` int(11) NOT NULL DEFAULT '0',
                     `is_recursive` tinyint(1) NOT NULL DEFAULT '0',
                     `access_rights` tinyint(1) NOT NULL DEFAULT '1',
                     `requesttype` int(11) NOT NULL DEFAULT '$requesttype',
                     `name` varchar(255) COLLATE utf8_unicode_ci NOT NULL,
                     `description` varchar(255) COLLATE utf8_unicode_ci,
                     `content` longtext COLLATE utf8_unicode_ci,
                     `plugin_formcreator_categories_id` tinyint(3) UNSIGNED NOT NULL,
                     `is_active` tinyint(1) NOT NULL DEFAULT '0',
                     `language` varchar(5) COLLATE utf8_unicode_ci NOT NULL,
                     `helpdesk_home` tinyint(1) NOT NULL DEFAULT '0',
                     `is_deleted` tinyint(1) NOT NULL DEFAULT '0',
                     `validation_required` tinyint(1) NOT NULL DEFAULT '0'
                  )
                  ENGINE = MyISAM
                  DEFAULT CHARACTER SET = utf8
                  COLLATE = utf8_unicode_ci;";
```

**Pull request merged**